frontend/src/components/explorer/MapViewer.vue

```diff
@@ -161,6 +168,9 @@ export default {
     this.loadMapFromParams();
   },
   methods: {
+    dimensionalState(showing2D) {
+      return showing2D ? '2d' : '3d';
+    },
```

Comment:
This is a good refactor and I think we could even go one step further and have it in a Vuex getter. Then we wouldn't need to pass it as a param. Not necessary now though.